### PR TITLE
ZotKat Issues

### PR DESCRIPTION
**Pull request by mara-spiess**

### CODE REVIEW

File: ubtue_AUP.js [hunk 1]
```diff
@@ -60,23 +60,80 @@ function getSearchResults(doc) {
...
+			let orcidAuthorName = orcidAuthor.textContent.trim();
+
+			if (i < orcidLinks.length) {
+				let orcidLink = orcidLinks[i];
+				let orcidUrl = orcidLink.getAttribute('href');
+				let match = orcidUrl.match(/https:\/\/orcid\.org\/(.+)/);
+				if (match) {
+					let orcidNumber = match[1];
+					item.notes.push("orcid: " + orcidNumber + ' | ' + orcidAuthorName + ' | ' + 'taken from website');
+				}
```
Comment (jriedl) [inline ubtue_AUP.js:78]: A bit tricky in this context, but in principle an approach that starts with one element and extract the other elements relative to this on is preferrable:
```js
function getOrcids(doc, item) {
    let authors = ZU.xpath(doc, '//li[contains(@class,"data-author")]//a[@class="nonDisambigAuthorLink"]');
    for (author of authors) {
        let authorName = ZU.xpathText(author, '.');
        if (!authorName)
            continue;
        let orcidUrl = ZU.xpath(author, './following-sibling::a/@href');
        if (!orcidUrl)
             continue;
        let orcidNumber = orcidUrl[0].value.match(/\d+-\d+-\d+-\d+x?/i);
        if (!orcidNumber)
             continue;
	item.notes.push("orcid: " + orcidNumber + ' | ' + authorName.trim() + ' | ' + 'taken from website');
    }
}
```

Comment (mara-spiess) [reply ubtue_AUP.js:78]: This version only works for me if an Orcid is present, which isn't always the case

Comment (jriedl) [reply ubtue_AUP.js:78]: `if (!orcidUrl)` has to be replaced by `if (!orcidUrl.length)` or, as this behaviour in contraction to the documentation (https://www.zotero.org/support/dev/translators/coding), even more cautiously `if (!orcidUrl || !orcidUrl.length)`.

File: ubtue_Herder.js [hunk 2]
```diff
@@ -140,6 +140,10 @@ function invokeEmbeddedMetadataTranslator(doc, url) {
 			item.ISSN = "2628-5762";
 			item.volume = item.issue;
 			item.issue = "";
+			if (item.title.includes("Buchvorstellungen")) {
```
Comment (jriedl) [inline ubtue_Herder.js:143]: Case insensitive to be more resilient?

File: ubtue_Herder.js [hunk 3]
```diff
@@ -140,6 +140,10 @@ function invokeEmbeddedMetadataTranslator(doc, url) {
...
+				item.tags.push('RezensionstagPica');
+				item.volume = item.volume.replace('buchvorstellungen/', '');
```
Comment (jriedl) [inline ubtue_Herder.js:145]: case insensitive to be more resilient?

**Pull request merged**